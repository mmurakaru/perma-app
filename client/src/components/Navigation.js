import styled from 'styled-components/macro'
import { ReactComponent as Spaces } from '../assets/spaces.svg'
import { ReactComponent as Overview } from '../assets/overview.svg'
import { ReactComponent as Home } from '../assets/home.svg'
import { NavLink } from 'react-router-dom'

export default function Navigation() {
    return (
        <NavigationContainer>
            <NavLink
                className="link"
                exact
                to="/"
                activeStyle={{
                    fill: 'var(--dark-green)',
                    color: 'var(--dark-green)',
                }}
            >
                <LinkWrapper>
                    <HomeIcon />
                    <LinkText>Home</LinkText>
                </LinkWrapper>
            </NavLink>
            <NavLink
                className="link"
                activeStyle={{
                    fill: 'var(--dark-green)',
                    color: 'var(--dark-green)',
                }}
                to="/Overview"
            >
                <LinkWrapper>
                    <OverviewIcon />
                    <LinkText>Add plant</LinkText>
                </LinkWrapper>
            </NavLink>
            <NavLink
                className="link"
                to="/Spaces"
                activeStyle={{
                    fill: 'var(--dark-green)',
                    color: 'var(--dark-green)',
                }}
            >
                <LinkWrapper>
                    <SpacesIcon />
                    <LinkText>Spaces</LinkText>
                </LinkWrapper>
            </NavLink>
        </NavigationContainer>
    )
}

const NavigationContainer = styled.nav`
    background-color: var(--medium-green);
    bottom: 0;
    display: flex;
    fill: var(--white);
    justify-content: space-around;
    height: 8%;
    position: fixed;
    padding: 10px 20px;
    width: 100%;
    max-width: 450px;

    .link {
        text-decoration: none;
        color: var(--white);
    }
`
const SpacesIcon = styled(Spaces)`
    width: 25px;
`
const OverviewIcon = styled(Overview)`
    width: 25px;
`
const HomeIcon = styled(Home)`
    width: 25px;
`

const LinkText = styled.span`
    font-size: 8px;
`
const LinkWrapper = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
`
