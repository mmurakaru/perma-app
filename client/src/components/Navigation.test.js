import { MemoryRouter as Router } from 'react-router-dom'
import { render } from '@testing-library/react'
import Navigation from './Navigation'
import 'jest-styled-components'

describe('Navigation', () => {
    it('has links to all pages', () => {
        const urls = ['/', '/Overview', '/Spaces']

        const { getAllByRole } = render(
            <Router>
                <Navigation />
            </Router>
        )

        getAllByRole('link').forEach((link) => {
            expect(urls).toContain(link.getAttribute('href'))
        })
    })

    it('renders correctly', () => {
        const { container } = render(
            <Router>
                <Navigation />
            </Router>
        )
        expect(container.firstChild).toMatchSnapshot()
    })
})
