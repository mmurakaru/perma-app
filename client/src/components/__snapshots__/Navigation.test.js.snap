// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
.c0 {
  background-color: #cfd8c6;
  border-top-left-radius: 25px;
  border-top-right-radius: 25px;
  position: fixed;
  bottom: 0;
  height: 8%;
  width: 100%;
  max-width: 450px;
  padding: 10px 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  fill: var(--white);
}

.c2 {
  width: 30px;
}

.c1 {
  width: 30px;
}

<section
  class="c0"
>
  <a
    aria-current="page"
    class="active"
    href="/"
    style="fill: var(--dark-green);"
  >
    <svg
      class="c1"
    >
      overview.svg
    </svg>
  </a>
  <a
    href="/Spaces"
  >
    <svg
      class="c2"
    >
      spaces.svg
    </svg>
  </a>
</section>
`;
